frontend/components/book-table-page/BooktablePage.tsx

```diff
@@ -0,0 +1,223 @@
+import { ControlledInput } from '@/components/common/ControlledInput';
+import { ControlledTestArea } from '@/components/common/ControlledTestArea';
+import { ControlledSelect } from '@/components/common/ControlledSelect';
+import { Button } from '@chakra-ui/react';
+import { useForm } from 'react-hook-form';
+import { InputsContainer } from '@/components/Take-away-page/component/InputsContainer';
+import { useState } from 'react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as zod from 'zod';
+import VerifyOtpModal from '@/components/book-table-page/VerifyOtpModal';
+import useTimer from './useTimer'; 
+import { Restaurant } from '@/types';
+import { isValidTime } from './timeUtils';
+import clsx from 'clsx';
+import { sanityClient } from '@/lib/sanityClient';
+
+
+interface BooktablePageProps {
+  restaurant: Restaurant;
+}
+
+type FormData = {
+  name: string;
+  phone: string;
+  time: string;
+  guests: string;
+  preference: string;
+  notes: string;
+  email: string;
+  date: string;
+};
+
+export function BooktablePage({ restaurant }: BooktablePageProps) {
+  
+  const requiredField = zod.string().min(1, { message: 'Required Field' });
+  const phoneSchema = zod.string()
+    .min(1, { message: 'Required Field' }) 
+    .regex(/^\d{9}$/, { message: 'Phone number invalid' });
+  
+  const FormDataSchema = zod.object({
+    name: requiredField,
+    phone: phoneSchema,
+    date: requiredField,
+    time: requiredField,
+    guests: requiredField,
+    email: requiredField.email(),
+    preference: zod.string(),
+    notes: zod.string(),
+  }).superRefine((data, context) => {
+    const isTimeValid = isValidTime(data.date, data.time, restaurant.Weekdaytime, restaurant.Weekandtime); 
+  
+    if (!isTimeValid) {
+      context.addIssue({
+        code: zod.ZodIssueCode.custom,
+        message: 'Time is outside of restaurant operating hours',
+        path: ['time'], 
+      });
+    }
+  
+    if (restaurant.blacklist.includes(data.phone)) {
+      context.addIssue({
+        code: zod.ZodIssueCode.custom,
+        message: 'Internal error, please try again later',
+        path: ['phone'],
+      });
+    }
+  });
+
+
+  const {
+    control,
+    handleSubmit,
+    trigger,
+    watch,
+  } = useForm<FormData>({
+    defaultValues: {
+      name: '',
+      phone: '',
+      time: '',
+      guests: '2',
+      preference: 'No-preference',
+      notes: '',
+      date: '',
+      email: '',
+    },
+    resolver: zodResolver(FormDataSchema),
+  });
+
+  const selectedDate = watch('date');
+
+  const [otp, setOtp] = useState(''); 
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [verifyOtp, setVerifyOtp] = useState(false);
+  const { timeLeft, isRunning, startTimer } = useTimer(60); 
+  
+  const onSubmit = async (data: FormData) => {
+    if (verifyOtp) {
+        try {
+            await sanityClient.create({
```

Comment:
Contact boris to add the token to the env otherwise we won't have access on the uat produciton